### PR TITLE
Add support for generating API keys in apps and for deleting existing keys

### PR DESCRIPTION
**Pull request by labkey-susanh**
#### Rationale
For LKSM Prof+ clients, we support usage of APIs, so we want to bring the ability to generate and delete API keys into the applications.  Here we replace the existing JSP for generating API tokens with the react component also used in the apps, which brings with it the ability to delete existing API keys.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/1385
* https://github.com/LabKey/testAutomation/pull/1778

#### Changes
* Add ProductFeature.apiKeys 
* Replace `apiKey.jsp` with `apiKeys` react entry point
* Add a UserApiKeysTableInfo, a filtered table over the existing APIKeysTable to the Core schema, but not to the Schema browser

### CODE REVIEW

File: core/src/org/labkey/core/security/SecurityController.java [hunk 1]
```diff
@@ -104,6 +113,7 @@
 import org.labkey.core.user.SecurityAccessView;
 import org.labkey.core.user.UserController;
 import org.labkey.core.user.UserController.AccessDetailRow;
+import org.springframework.dao.DataIntegrityViolationException;
```
Comment (labkey-adam) [inline core/src/org/labkey/core/security/SecurityController.java]: Just a bunch of unused imports?

File: core/src/org/labkey/core/query/CoreQuerySchema.java [hunk 2]
```diff
@@ -161,6 +167,8 @@ public TableInfo createTable(String name, ContainerFilter cf)
             return getDataStatesTable();
         if (API_KEYS_TABLE_NAME.equalsIgnoreCase(name) && getUser().hasRootPermission(UserManagementPermission.class))
             return new ApiKeysTableInfo(this);
+        if (USER_API_KEYS_TABLE_NAME.equalsIgnoreCase(name))
```
Comment (labkey-adam) [inline core/src/org/labkey/core/query/CoreQuerySchema.java:171]: Optional: could add a comment we're intentionally not adding this to schema browser tree

**Review by labkey-adam — APPROVED** [review]

File: core/src/client/APIKeys/APIKeys.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+import React, { FC } from 'react';
+import { APIKeysPanel, AppContextProvider, ServerContextProvider, withAppUser, } from '@labkey/components';
+import { getServerContext } from '@labkey/api';
+
+import '../DomainDesigner.scss';
+
+export const App: FC<any> = () => {
+    const serverContext = withAppUser(getServerContext());
```
Comment (cnathe) [inline core/src/client/APIKeys/APIKeys.tsx:23]: glad this pattern worked and wasn't too much overhead

File: core/src/client/entryPoints.js [hunk 4]
```diff
@@ -74,5 +74,9 @@ module.exports = {
         title: 'Query Select Input',
         generateLib: true, // used in TypeAheadSelectDisplayColumn.java
         path: './src/client/QuerySelectInput'
+    }, {
+        name: 'apiKeys', // used in ApiKeyViewProvider.java
```
Comment (cnathe) [inline core/src/client/entryPoints.js:78]: shouldn't this entryPoint config set `permission: 'admin'`? 
The main usage is in the ApiKeyViewProvider.java which likely checks the perm (haven't look yet) but a user could still technically access this page via the /views.

Comment (labkey-susanh) [reply core/src/client/entryPoints.js:78]: No. Everyone gets to look at their own API keys and generate them if enabled.

File: core/src/org/labkey/core/security/SecurityController.java (hunk 1 above)
Comment (labkey-susanh) [reply core/src/org/labkey/core/security/SecurityController.java]: Removed.

File: core/src/org/labkey/core/query/CoreQuerySchema.java (hunk 2 above)
Comment (labkey-susanh) [reply core/src/org/labkey/core/query/CoreQuerySchema.java:171]: Done.

**Pull request merged**